pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -561,3 +564,34 @@ func (c *Cluster) resolveNode(nodeName string) *rookalpha.Node {
 	rookNode.Resources = k8sutil.MergeResourceRequirements(rookNode.Resources, c.resources)
 	return rookNode
 }
+
+func (c *Cluster) getValidStorageNodes() []rookalpha.Node {
+	storageNodes := c.Storage.Nodes
```

Comment:
We expect that `c.Storage.Nodes` is the list of nodes from the crd, right? Will this be reset upon each iteration of the osd orchestration? I noticed above that c.Storage.Nodes was set to the result of this function call. Will that potentially limit the nodes in the next orchestration? Or is the nodes list reset somewhere else each time the crd change triggers a new orchestration?

Reply:
yes, it is reset after update. This fix will restrict storage nodes to limit to affinity policy after they are updated in the CRD.